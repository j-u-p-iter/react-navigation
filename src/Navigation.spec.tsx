import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import { cleanup, fireEvent, render } from "react-testing-library";

import { Navigation } from "./.";

const config = [
  {
    id: "/path/1",
    title: "Title 1",
    children: [
      {
        id: "/path/2",
        title: "Title 2",
        children: [
          {
            id: "/path/3",
            title: "Title 3",
            children: [
              {
                id: "/path/4",
                title: "Title 4",
                children: [
                  {
                    id: "/path/5",
                    title: "Title 5"
                  },
                  {
                    id: "/path/6",
                    title: "Title 6"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  },
  {
    id: "/path/7",
    title: "Title 7",
    children: [
      {
        id: "/path/8",
        title: "Title 8",
        children: [
          {
            id: "/path/9",
            title: "Title 9"
          }
        ]
      }
    ]
  }
];

describe("Navigation", () => {
  let getNavigationEl: any;

  beforeAll(() => {
    getNavigationEl = (props: any = {}) => (
      <MemoryRouter>
        <Navigation
          config={config}
          currentPath="/path/4"
          activeClassName="active-class-name"
          className="navigation-class-name"
          {...props}
        />
      </MemoryRouter>
    );
  });

  afterEach(cleanup);

  it("renders correctly by default", () => {
    const { container } = render(getNavigationEl());
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with all props", () => {
    const { container } = render(
      getNavigationEl({ withChildrenClassName: "with-children-class-name" })
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("toggles items correctly", () => {
    const { container, getByText } = render(
      getNavigationEl({ withChildrenClassName: "with-children-class-name" })
    );

    fireEvent.click(getByText("Title 2"));

    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(getByText("Title 1"));

    expect(container.firstChild).toMatchSnapshot();
  });

  it("doesn't toggle items without children", () => {
    const { container, getByText } = render(
      getNavigationEl({ withChildrenClassName: "with-children-class-name" })
    );

    fireEvent.click(getByText("Title 5"));

    expect(container.firstChild).toMatchSnapshot();
  });
});
