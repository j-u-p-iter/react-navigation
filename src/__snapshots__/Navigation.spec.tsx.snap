// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation doesn't toggle items without children 1`] = `
<nav
  class="navigation-class-name"
>
  <ul>
    <li
      class="active-class-name with-children-class-name"
      id="/path/1"
    >
      <a
        href="/path/1"
      >
        Title 1
      </a>
      <ul>
        <li
          class="active-class-name with-children-class-name"
          id="/path/2"
        >
          <a
            href="/path/2"
          >
            Title 2
          </a>
          <ul>
            <li
              class="active-class-name with-children-class-name"
              id="/path/3"
            >
              <a
                href="/path/3"
              >
                Title 3
              </a>
              <ul>
                <li
                  class="with-children-class-name"
                  id="/path/4"
                >
                  <a
                    href="/path/4"
                  >
                    Title 4
                  </a>
                  <ul>
                    <li
                      class=""
                      id="/path/5"
                    >
                      <a
                        href="/path/5"
                      >
                        Title 5
                      </a>
                    </li>
                    <li
                      class=""
                      id="/path/6"
                    >
                      <a
                        href="/path/6"
                      >
                        Title 6
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li
      class="with-children-class-name"
      id="/path/7"
    >
      <a
        href="/path/7"
      >
        Title 7
      </a>
      <ul>
        <li
          class="with-children-class-name"
          id="/path/8"
        >
          <a
            href="/path/8"
          >
            Title 8
          </a>
          <ul>
            <li
              class=""
              id="/path/9"
            >
              <a
                href="/path/9"
              >
                Title 9
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;

exports[`Navigation renders correctly by default 1`] = `
<nav
  class="navigation-class-name"
>
  <ul>
    <li
      class="active-class-name "
      id="/path/1"
    >
      <a
        href="/path/1"
      >
        Title 1
      </a>
      <ul>
        <li
          class="active-class-name "
          id="/path/2"
        >
          <a
            href="/path/2"
          >
            Title 2
          </a>
          <ul>
            <li
              class="active-class-name "
              id="/path/3"
            >
              <a
                href="/path/3"
              >
                Title 3
              </a>
              <ul>
                <li
                  class="active-class-name "
                  id="/path/4"
                >
                  <a
                    href="/path/4"
                  >
                    Title 4
                  </a>
                  <ul>
                    <li
                      class=""
                      id="/path/5"
                    >
                      <a
                        href="/path/5"
                      >
                        Title 5
                      </a>
                    </li>
                    <li
                      class=""
                      id="/path/6"
                    >
                      <a
                        href="/path/6"
                      >
                        Title 6
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li
      class=""
      id="/path/7"
    >
      <a
        href="/path/7"
      >
        Title 7
      </a>
      <ul>
        <li
          class=""
          id="/path/8"
        >
          <a
            href="/path/8"
          >
            Title 8
          </a>
          <ul>
            <li
              class=""
              id="/path/9"
            >
              <a
                href="/path/9"
              >
                Title 9
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;

exports[`Navigation renders correctly with all props 1`] = `
<nav
  class="navigation-class-name"
>
  <ul>
    <li
      class="active-class-name with-children-class-name"
      id="/path/1"
    >
      <a
        href="/path/1"
      >
        Title 1
      </a>
      <ul>
        <li
          class="active-class-name with-children-class-name"
          id="/path/2"
        >
          <a
            href="/path/2"
          >
            Title 2
          </a>
          <ul>
            <li
              class="active-class-name with-children-class-name"
              id="/path/3"
            >
              <a
                href="/path/3"
              >
                Title 3
              </a>
              <ul>
                <li
                  class="active-class-name with-children-class-name"
                  id="/path/4"
                >
                  <a
                    href="/path/4"
                  >
                    Title 4
                  </a>
                  <ul>
                    <li
                      class=""
                      id="/path/5"
                    >
                      <a
                        href="/path/5"
                      >
                        Title 5
                      </a>
                    </li>
                    <li
                      class=""
                      id="/path/6"
                    >
                      <a
                        href="/path/6"
                      >
                        Title 6
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li
      class="with-children-class-name"
      id="/path/7"
    >
      <a
        href="/path/7"
      >
        Title 7
      </a>
      <ul>
        <li
          class="with-children-class-name"
          id="/path/8"
        >
          <a
            href="/path/8"
          >
            Title 8
          </a>
          <ul>
            <li
              class=""
              id="/path/9"
            >
              <a
                href="/path/9"
              >
                Title 9
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;

exports[`Navigation toggles items correctly 1`] = `
<nav
  class="navigation-class-name"
>
  <ul>
    <li
      class="active-class-name with-children-class-name"
      id="/path/1"
    >
      <a
        href="/path/1"
      >
        Title 1
      </a>
      <ul>
        <li
          class="with-children-class-name"
          id="/path/2"
        >
          <a
            href="/path/2"
          >
            Title 2
          </a>
          <ul>
            <li
              class="active-class-name with-children-class-name"
              id="/path/3"
            >
              <a
                href="/path/3"
              >
                Title 3
              </a>
              <ul>
                <li
                  class="active-class-name with-children-class-name"
                  id="/path/4"
                >
                  <a
                    href="/path/4"
                  >
                    Title 4
                  </a>
                  <ul>
                    <li
                      class=""
                      id="/path/5"
                    >
                      <a
                        href="/path/5"
                      >
                        Title 5
                      </a>
                    </li>
                    <li
                      class=""
                      id="/path/6"
                    >
                      <a
                        href="/path/6"
                      >
                        Title 6
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li
      class="with-children-class-name"
      id="/path/7"
    >
      <a
        href="/path/7"
      >
        Title 7
      </a>
      <ul>
        <li
          class="with-children-class-name"
          id="/path/8"
        >
          <a
            href="/path/8"
          >
            Title 8
          </a>
          <ul>
            <li
              class=""
              id="/path/9"
            >
              <a
                href="/path/9"
              >
                Title 9
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;

exports[`Navigation toggles items correctly 2`] = `
<nav
  class="navigation-class-name"
>
  <ul>
    <li
      class="with-children-class-name"
      id="/path/1"
    >
      <a
        href="/path/1"
      >
        Title 1
      </a>
      <ul>
        <li
          class="with-children-class-name"
          id="/path/2"
        >
          <a
            href="/path/2"
          >
            Title 2
          </a>
          <ul>
            <li
              class="active-class-name with-children-class-name"
              id="/path/3"
            >
              <a
                href="/path/3"
              >
                Title 3
              </a>
              <ul>
                <li
                  class="active-class-name with-children-class-name"
                  id="/path/4"
                >
                  <a
                    href="/path/4"
                  >
                    Title 4
                  </a>
                  <ul>
                    <li
                      class=""
                      id="/path/5"
                    >
                      <a
                        href="/path/5"
                      >
                        Title 5
                      </a>
                    </li>
                    <li
                      class=""
                      id="/path/6"
                    >
                      <a
                        href="/path/6"
                      >
                        Title 6
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li
      class="with-children-class-name"
      id="/path/7"
    >
      <a
        href="/path/7"
      >
        Title 7
      </a>
      <ul>
        <li
          class="with-children-class-name"
          id="/path/8"
        >
          <a
            href="/path/8"
          >
            Title 8
          </a>
          <ul>
            <li
              class=""
              id="/path/9"
            >
              <a
                href="/path/9"
              >
                Title 9
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;
